src/state/action-creators.js

```diff
@@ -61,47 +64,66 @@ export type CompleteLiftAction = {|
     id: DraggableId,
     client: InitialDragPositions,
     windowScroll: Position,
-    isScrollAllowed: boolean,
+    autoScrollMode: AutoScrollMode,
   |}
 |}
 
 export const completeLift = (
   id: DraggableId,
   client: InitialDragPositions,
   windowScroll: Position,
-  isScrollAllowed: boolean,
+  autoScrollMode: AutoScrollMode,
 ): CompleteLiftAction => ({
   type: 'COMPLETE_LIFT',
   payload: {
     id,
     client,
     windowScroll,
-    isScrollAllowed,
+    autoScrollMode,
   },
 });
 
-export type PublishDraggableDimensionsAction = {|
-  type: 'PUBLISH_DRAGGABLE_DIMENSIONS',
-  payload: DraggableDimension[]
+export type PublishDraggableDimensionAction = {|
+  type: 'PUBLISH_DRAGGABLE_DIMENSION',
+  payload: DraggableDimension,
 |}
 
-export const publishDraggableDimensions =
-  (dimensions: DraggableDimension[]): PublishDraggableDimensionsAction => ({
-    type: 'PUBLISH_DRAGGABLE_DIMENSIONS',
-    payload: dimensions,
+export const publishDraggableDimension =
+  (dimension: DraggableDimension): PublishDraggableDimensionAction => ({
+    type: 'PUBLISH_DRAGGABLE_DIMENSION',
+    payload: dimension,
   });
 
-export type PublishDroppableDimensionsAction = {|
-  type: 'PUBLISH_DROPPABLE_DIMENSIONS',
-  payload: DroppableDimension[]
+export type PublishDroppableDimensionAction = {|
+  type: 'PUBLISH_DROPPABLE_DIMENSION',
+  payload: DroppableDimension,
 |}
 
-export const publishDroppableDimensions =
-  (dimensions: DroppableDimension[]): PublishDroppableDimensionsAction => ({
-    type: 'PUBLISH_DROPPABLE_DIMENSIONS',
-    payload: dimensions,
+export const publishDroppableDimension =
+  (dimension: DroppableDimension): PublishDroppableDimensionAction => ({
+    type: 'PUBLISH_DROPPABLE_DIMENSION',
+    payload: dimension,
   });
 
+export type BulkPublishDimensionsAction = {|
+  type: 'BULK_DIMENSION_PUBLISH',
+  payload: {|
+    droppables: DroppableDimension[],
+    draggables: DraggableDimension[],
+  |}
+|}
+
+export const bulkPublishDimensions = (
```

Comment:
now one big action to bulk publish dimensions. This gets around a timing issue for `onDragUpdate`